### DISCUSSION TITLE
How to set an environment variable with a value composed with an endpoint address?

### DISCUSSION THREAD

**Discussion by paulomorgado** [microsoft/aspire#4874] [Q&A] [answered]
This gets tricky because, if the resource is a container resource, depending on the docker engine, it might need to change the host name.

It works for `WithReference`, but how would it work for any other environment variable when it needs to be composed with path?

**Comment by davidfowl** [comment, marked as answer]
[Does this help](https://github.com/dotnet/aspire/blob/e0552c0bbf3330c2694e72961a97941a5773e139/src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs#L123-L126)

There are lots of issues around this (small paper cuts):
- https://github.com/dotnet/aspire/issues/3735
- https://github.com/dotnet/aspire/issues/3937

**Reply by paulomorgado** [reply]
That's a good one. For the time being, I'm using connection strings for things like Keycloak realms to inject into `JwtBearerOptions.Authority`.
